src/Agent.Sdk/Util/PlatformUtil.cs

```diff
@@ -75,6 +85,36 @@ public static bool RunningOnRHEL6
             }
         }
 
+        public static string GetSystemId()
+        {
+            switch (PlatformUtil.HostOS)
+            {
+                case PlatformUtil.OS.Linux:
+                    return GetLinuxId();
+                case PlatformUtil.OS.OSX:
+                    return "MacOS";
+                case PlatformUtil.OS.Windows:
+                    return GetWindowsId();
+                default:
+                    return null;
+            }
+        }
+
+        public static OSVersion GetSystemVersion()
```

Comment:
This can be simplified a bit
```
        public static OSVersion GetSystemVersion()
        {
            return PlatformUtil.HostOS switch
            {
                PlatformUtil.OS.Linux => new OSVersion(GetLinuxName(), null),
                PlatformUtil.OS.OSX => new OSVersion(GetOSxName(), null),
                PlatformUtil.OS.Windows => new OSVersion(GetWindowsName(), GetWindowsVersion()),
                _ => null
            };
        }
```